diff --git a/Makefile b/Makefile
index 3c0b21f..bd36c29 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 #
-# $FreeBSD$
+# $FreeBSD: stable/9/Makefile 252588 2013-07-03 19:12:35Z obrien $
 #
 # The user-driven targets are:
 #
@@ -31,10 +31,6 @@
 # targets             - Print a list of supported TARGET/TARGET_ARCH pairs
 #                       for world and kernel targets.
 # toolchains          - Build a toolchain for all world and kernel targets.
-# xdev                - xdev-build + xdev-install for the architecture
-#                       specified with XDEV and XDEV_ARCH.
-# xdev-build          - Build cross-development tools.
-# xdev-install        - Install cross-development tools.
 #
 # This makefile is simple by design. The FreeBSD make automatically reads
 # the /usr/share/mk/sys.mk unless the -m argument is specified on the
@@ -113,15 +109,13 @@ TGTS+=	${BITGTS}
 .ORDER: buildworld installworld
 .ORDER: buildworld distributeworld
 .ORDER: buildworld buildkernel
-.ORDER: installworld distribution
-.ORDER: installworld installkernel
 .ORDER: buildkernel installkernel
 .ORDER: buildkernel installkernel.debug
 .ORDER: buildkernel reinstallkernel
 .ORDER: buildkernel reinstallkernel.debug
 
 PATH=	/sbin:/bin:/usr/sbin:/usr/bin
-MAKEOBJDIRPREFIX?=	/usr/obj
+MAKEOBJDIRPREFIX?=	${.CURDIR}/obj	
 _MAKEOBJDIRPREFIX!= /usr/bin/env -i PATH=${PATH} ${MAKE} \
     ${.MAKEFLAGS:MMAKEOBJDIRPREFIX=*} __MAKE_CONF=${__MAKE_CONF} \
     -f /dev/null -V MAKEOBJDIRPREFIX dummy
@@ -224,17 +218,8 @@ cleanworld:
 ${TGTS}:
 	${_+_}@cd ${.CURDIR}; ${_MAKE} ${.TARGET}
 
-# The historic default "all" target creates files which may cause stale
-# or (in the cross build case) unlinkable results. Fail with an error
-# when no target is given. The users can explicitly specify "all"
-# if they want the historic behavior.
-.MAIN:	_guard
-
-_guard:
-	@echo
-	@echo "Explicit target required (use \"all\" for historic behavior)"
-	@echo
-	@false
+# Set a reasonable default
+.MAIN:	all
 
 STARTTIME!= LC_ALL=C date
 CHECK_TIME!= find ${.CURDIR}/sys/sys/param.h -mtime -0s ; echo
@@ -424,8 +409,7 @@ universe_kernels: universe_kernconfs
 TARGET!=	uname -m
 .endif
 KERNCONFS!=	cd ${KERNSRCDIR}/${TARGET}/conf && \
-		find [[:upper:][:digit:]]*[[:upper:][:digit:]] \
-		-type f -maxdepth 0 \
+		find [A-Z0-9]*[A-Z0-9] -type f -maxdepth 0 \
 		! -name DEFAULTS ! -name NOTES
 universe_kernconfs:
 .for kernel in ${KERNCONFS}
diff --git a/Makefile.inc1 b/Makefile.inc1
index 80ffaea..4e3998d 100644
--- a/Makefile.inc1
+++ b/Makefile.inc1
@@ -1,5 +1,5 @@
 #
-# $FreeBSD$
+# $FreeBSD: stable/9/Makefile.inc1 253164 2013-07-10 18:27:14Z brooks $
 #
 # Make command line options:
 #	-DNO_CLEANDIR run ${MAKE} clean, instead of ${MAKE} cleandir
@@ -62,10 +62,6 @@ TARGET_ARCH=${TARGET}
 # use that new version.  And the new (dynamically-linked) /bin/sh
 # will expect to find appropriate libraries in /lib and /libexec.
 #
-SRCDIR?=	${.CURDIR}
-.if defined(SUBDIR_OVERRIDE)
-SUBDIR=	${SUBDIR_OVERRIDE}
-.else
 SUBDIR=	share/info lib libexec
 SUBDIR+=bin
 .if ${MK_GAMES} != "no"
@@ -104,6 +100,9 @@ SUBDIR+=etc
 SUBDIR+= ${_DIR}
 .endif
 .endfor
+
+.if defined(SUBDIR_OVERRIDE)
+SUBDIR=		${SUBDIR_OVERRIDE}
 .endif
 
 .if defined(NOCLEAN)
@@ -120,7 +119,7 @@ BUILDENV_SHELL?=/bin/sh
 SVN?=		/usr/local/bin/svn
 SVNFLAGS?=	-r HEAD
 
-MAKEOBJDIRPREFIX?=	/usr/obj
+MAKEOBJDIRPREFIX?=	${.CURDIR}/obj
 .if !defined(OSRELDATE)
 .if exists(/usr/include/osreldate.h)
 OSRELDATE!=	awk '/^\#define[[:space:]]*__FreeBSD_version/ { print $$3 }' \
@@ -131,11 +130,8 @@ OSRELDATE=	0
 .endif
 
 .if !defined(VERSION)
-REVISION!=	${MAKE} -C ${SRCDIR}/release -V REVISION
-BRANCH!=	${MAKE} -C ${SRCDIR}/release -V BRANCH
-SRCRELDATE!=	awk '/^\#define[[:space:]]*__FreeBSD_version/ { print $$3 }' \
-		${SRCDIR}/sys/sys/param.h
-VERSION=	FreeBSD ${REVISION}-${BRANCH:C/-p[0-9]+$//} ${TARGET_ARCH} ${SRCRELDATE}
+VERSION!=	uname -srp
+VERSION+=	${OSRELDATE}
 .endif
 
 KNOWN_ARCHES?=	amd64 arm armeb/arm i386 i386/pc98 ia64 mipsel/mips mipseb/mips mips64el/mips mips64eb/mips mipsn32el/mips mipsn32eb/mips powerpc powerpc64/powerpc sparc64
@@ -234,7 +230,6 @@ CROSSENV+=	GROFF_BIN_PATH=${WORLDTMP}/legacy/usr/bin \
 		GROFF_TMAC_PATH=${WORLDTMP}/legacy/usr/share/tmac
 .endif
 
-_BOOTSTRAP_MAKEINFO?=	${MK_INFO}
 # bootstrap-tools stage
 BMAKEENV=	INSTALL="sh ${.CURDIR}/tools/install.sh" \
 		PATH=${BPATH}:${PATH} \
@@ -249,7 +244,6 @@ BMAKE=		MAKEOBJDIRPREFIX=${WORLDTMP} \
 		SSP_CFLAGS= \
 		-DWITHOUT_HTML -DWITHOUT_INFO -DNO_LINT -DWITHOUT_MAN \
 		-DNO_PIC -DWITHOUT_PROFILE -DNO_SHARED \
-		_BOOTSTRAP_MAKEINFO=${_BOOTSTRAP_MAKEINFO} \
 		-DNO_CPU_CFLAGS -DNO_WARNS -DNO_CTF -DEARLY_BUILD
 
 # build-tools stage
@@ -276,7 +270,7 @@ WMAKEENV=	${CROSSENV} \
 		PATH=${TMPPATH}
 
 # make hierarchy
-HMAKE=		PATH=${TMPPATH} ${MAKE} LOCAL_MTREE=${LOCAL_MTREE:Q}
+HMAKE=		PATH=${TMPPATH} ${MAKE} LOCAL_MTREE=${LOCAL_MTREE}
 .if defined(NO_ROOT)
 HMAKE+=		PATH=${TMPPATH} METALOG=${METALOG} -DNO_ROOT
 .endif
@@ -338,7 +332,6 @@ LIB32WMAKEENV+=	MAKEOBJDIRPREFIX=${OBJTREE}/lib32 \
 		PATH=${TMPPATH} \
 		LIBDIR=/usr/lib32 \
 		SHLIBDIR=/usr/lib32 \
-		LIBPRIVATEDIR=/usr/lib32/private \
 		COMPILER_TYPE=${WMAKE_COMPILER_TYPE}
 LIB32WMAKEFLAGS+=	\
 		CC="${CC} ${LIB32FLAGS}" \
@@ -562,10 +555,10 @@ build32:
 	    -DEARLY_BUILD build-tools
 .endfor
 	cd ${.CURDIR}; \
-	    ${LIB32WMAKE} -f Makefile.inc1 -DNO_FSCHG libraries
+	    ${LIB32WMAKE} -f Makefile.inc1 libraries
 .for _t in obj depend all
 	cd ${.CURDIR}/libexec/rtld-elf; PROG=ld-elf32.so.1 ${LIB32WMAKE} \
-	    -DNO_FSCHG DIRPRFX=libexec/rtld-elf/ ${_t}
+	    DIRPRFX=libexec/rtld-elf/ ${_t}
 	cd ${.CURDIR}/usr.bin/ldd; PROG=ldd32 ${LIB32WMAKE} \
 	    DIRPRFX=usr.bin/ldd ${_t}
 .endfor
@@ -623,11 +616,6 @@ buildworld_epilogue:
 buildenvvars:
 	@echo ${WMAKEENV:Q}
 
-.if ${.TARGETS:Mbuildenv}
-.if ${.MAKEFLAGS:M-j}
-.error The buildenv target is incompatible with -j
-.endif
-.endif
 buildenv:
 	@echo Entering world for ${TARGET_ARCH}:${TARGET}
 	@cd ${.CURDIR} && env ${WMAKEENV} ${BUILDENV_SHELL} || true
@@ -706,7 +694,7 @@ _nmtree_itools=	nmtree
 ITOOLS=	[ awk cap_mkdb cat chflags chmod chown \
 	date echo egrep find grep id install ${_install-info} \
 	ln lockf make mkdir mtree ${_nmtree_itools} mv pwd_mkdb \
-	rm sed sh strip sysctl test true uname wc ${_zoneinfo}
+	rm sed sh sysctl test true uname wc ${_zoneinfo}
 
 #
 # distributeworld
@@ -731,7 +719,6 @@ MTREE_MAGIC?=	mtree 2.0
 
 distributeworld installworld: installcheck installcheck_UGID
 	mkdir -p ${INSTALLTMP}
-	export PATH=${BPATH}:${PATH} ; \
 	progs=$$(for prog in ${ITOOLS}; do \
 		if progpath=`which $$prog`; then \
 			echo $$progpath; \
@@ -781,7 +768,7 @@ distributeworld installworld: installcheck installcheck_UGID
 	cd ${.CURDIR}/etc; ${CROSSENV} PATH=${TMPPATH} ${MAKE} \
 	    METALOG=${METALOG} ${IMAKE_INSTALL} ${IMAKE_MTREE} \
 	    DISTBASE=/base DESTDIR=${DESTDIR}/${DISTDIR}/base \
-	    LOCAL_MTREE=${LOCAL_MTREE:Q} distrib-dirs
+	    LOCAL_MTREE=${LOCAL_MTREE} distrib-dirs
 .endif
 	${_+_}cd ${.CURDIR}; ${IMAKE} re${.TARGET:S/world$//}; \
 	    ${IMAKEENV} rm -rf ${INSTALLTMP}
@@ -827,7 +814,7 @@ reinstall:
 	@echo ">>> Making hierarchy"
 	@echo "--------------------------------------------------------------"
 	${_+_}cd ${.CURDIR}; ${MAKE} -f Makefile.inc1 \
-	    LOCAL_MTREE=${LOCAL_MTREE:Q} hierarchy
+	    LOCAL_MTREE=${LOCAL_MTREE} hierarchy
 	@echo
 	@echo "--------------------------------------------------------------"
 	@echo ">>> Installing everything"
@@ -934,7 +921,7 @@ buildkernel:
 	cd ${KRNLCONFDIR}; \
 		PATH=${TMPPATH} \
 		    config ${CONFIGARGS} -d ${KRNLOBJDIR}/${_kernel} \
-			-I '${KERNCONFDIR}' '${KERNCONFDIR}/${_kernel}'
+			${KERNCONFDIR}/${_kernel}
 .endif
 .if !defined(NO_CLEAN) && !defined(NO_KERNELCLEAN)
 	@echo
@@ -957,16 +944,15 @@ buildkernel:
 	cd ${KRNLOBJDIR}/${_kernel}; \
 	    PATH=${BPATH}:${PATH} \
 	    MAKESRCPATH=${KERNSRCDIR}/dev/aic7xxx/aicasm \
-	    ${MAKE} SSP_CFLAGS= -DNO_CPU_CFLAGS -DNO_WARNS -DNO_CTF \
-	    -DEARLY_BUILD -f ${KERNSRCDIR}/dev/aic7xxx/aicasm/Makefile
+	    ${MAKE} SSP_CFLAGS= -DNO_CPU_CFLAGS -DNO_CTF -DEARLY_BUILD \
+	    -f ${KERNSRCDIR}/dev/aic7xxx/aicasm/Makefile
 # XXX - Gratuitously builds aicasm in the ``makeoptions NO_MODULES'' case.
 .if !defined(MODULES_WITH_WORLD) && !defined(NO_MODULES) && exists(${KERNSRCDIR}/modules)
 .for target in obj depend all
 	cd ${KERNSRCDIR}/modules/aic7xxx/aicasm; \
 	    PATH=${BPATH}:${PATH} \
 	    MAKEOBJDIRPREFIX=${KRNLOBJDIR}/${_kernel}/modules \
-	    ${MAKE} SSP_CFLAGS= -DNO_CPU_CFLAGS -DNO_WARNS -DNO_CTF \
-	    -DEARLY_BUILD ${target}
+	    ${MAKE} SSP_CFLAGS= -DNO_CPU_CFLAGS -DNO_CTF -DEARLY_BUILD ${target}
 .endfor
 .endif
 .if !defined(NO_KERNELDEPEND)
@@ -1094,8 +1080,6 @@ legacy:
 #
 # bootstrap-tools: Build tools needed for compatibility
 #
-_bt=		_bootstrap-tools	
-
 .if ${MK_GAMES} != "no"
 _strfile=	games/fortune/strfile
 .endif
@@ -1127,15 +1111,11 @@ _lex=		usr.bin/lex
 .if ${BOOTSTRAPPING} < 900006 || ${BOOTSTRAPPING} >= 1000013
 _yacc=		lib/liby \
 		usr.bin/yacc
-
-${_bt}-usr.bin/yacc: ${_bt}-lib/liby
 .endif
 
 .if ${BOOTSTRAPPING} < 1000026
 _nmtree=	lib/libnetbsd \
 		usr.sbin/nmtree
-
-${_bt}-usr.sbin/nmtree: ${_bt}-lib/libnetbsd
 .endif
 
 .if ${BOOTSTRAPPING} < 1000027
@@ -1162,39 +1142,24 @@ _clang_tblgen= \
 	lib/clang/libllvmtablegen \
 	usr.bin/clang/tblgen \
 	usr.bin/clang/clang-tblgen
-
-${_bt}-usr.bin/clang/clang-tblgen: ${_bt}-lib/clang/libllvmtablegen ${_bt}-lib/clang/libllvmsupport
-${_bt}-usr.bin/clang/tblgen: ${_bt}-lib/clang/libllvmtablegen ${_bt}-lib/clang/libllvmsupport
 .endif
 
 .if ${MK_CDDL} != "no" && \
     ((${BOOTSTRAPPING} < 1000034 && \
           !(${BOOTSTRAPPING} >= 901505 && ${BOOTSTRAPPING} < 999999)) \
       || (${MACHINE} != ${TARGET} || ${MACHINE_ARCH} != ${TARGET_ARCH}))
-_dtrace_tools= cddl/lib/libctf lib/libelf \
+_dtrace_tools= cddl/usr.bin/sgsmsg cddl/lib/libctf lib/libelf \
     lib/libdwarf cddl/usr.bin/ctfconvert cddl/usr.bin/ctfmerge
-
-${_bt}-cddl/usr.bin/ctfconvert: ${_bt}-lib/libelf ${_bt}-lib/libdwarf ${_bt}-cddl/lib/libctf
-${_bt}-cddl/usr.bin/ctfmerge: ${_bt}-lib/libelf ${_bt}-lib/libdwarf ${_bt}-cddl/lib/libctf
 .endif
 
 .if ${MK_FDT} != "no"
 _dtc= gnu/usr.bin/dtc
 .endif
 
-.if ${_BOOTSTRAP_MAKEINFO} != "no"
-_texinfo=	gnu/usr.bin/texinfo/libtxi \
-		gnu/usr.bin/texinfo/makeinfo \
-		gnu/usr.bin/texinfo/install-info
-
-${_bt}-gnu/usr.bin/texinfo/install-info: ${_bt}-gnu/usr.bin/texinfo/libtxi
-${_bt}-gnu/usr.bin/texinfo/makeinfo: ${_bt}-gnu/usr.bin/texinfo/libtxi
-.endif
-bootstrap-tools: .PHONY
-
 #	Please document (add comment) why something is in 'bootstrap-tools'.
 #	Try to bound the building of the bootstrap-tool to just the
 #	FreeBSD versions that need the tool built at this stage of the build.
+bootstrap-tools:
 .for _tool in \
     ${_clang_tblgen} \
     ${_dtrace_tools} \
@@ -1212,21 +1177,18 @@ bootstrap-tools: .PHONY
     ${_sed} \
     ${_yacc} \
     ${_lex} \
+    lib/libmd \
     usr.bin/xinstall \
     ${_gensnmptree} \
     usr.sbin/config \
     ${_crunchgen} \
-    ${_nmtree} \
-    ${_texinfo}
-${_bt}-${_tool}: .PHONY .MAKE
+    ${_nmtree}
 	${_+_}@${ECHODIR} "===> ${_tool} (obj,depend,all,install)"; \
 		cd ${.CURDIR}/${_tool}; \
 		${MAKE} DIRPRFX=${_tool}/ obj; \
 		${MAKE} DIRPRFX=${_tool}/ depend; \
 		${MAKE} DIRPRFX=${_tool}/ all; \
 		${MAKE} DIRPRFX=${_tool}/ DESTDIR=${MAKEOBJDIRPREFIX}/legacy install
-
-bootstrap-tools: ${_bt}-${_tool}
 .endfor
 
 #
@@ -1240,7 +1202,7 @@ _aicasm= sys/modules/aic7xxx/aicasm
 _share=	share/syscons/scrnmaps
 .endif
 
-.if ${MK_GCC} != "no"
+.if ${MK_GCC} != "no" && (${MK_CLANG_IS_CC} == "no" || ${TARGET} == "pc98")
 _gcc_tools= gnu/usr.bin/cc/cc_tools
 .endif
 
@@ -1281,7 +1243,6 @@ build-tools:
 		${MAKE} DIRPRFX=${_tool}/ all
 .endfor
 
-
 #
 # cross-tools: Build cross-building tools
 #
@@ -1390,22 +1351,17 @@ _prebuild_libs=	${_kerberos5_lib_libasn1} ${_kerberos5_lib_libhdb} \
 		lib/libkiconv lib/libkvm lib/liblzma lib/libmd \
 		lib/ncurses/ncurses lib/ncurses/ncursesw \
 		lib/libopie lib/libpam ${_lib_libthr} \
-		${_lib_libradius} lib/libsbuf lib/libtacplus \
+		lib/libradius lib/libsbuf lib/libtacplus \
 		${_cddl_lib_libumem} ${_cddl_lib_libnvpair} \
-		${_cddl_lib_libavl} \
 		${_cddl_lib_libzfs_core} \
 		lib/libutil ${_lib_libypclnt} lib/libz lib/msun \
-		${_secure_lib_libcrypto} ${_lib_libldns} \
-		${_secure_lib_libssh} ${_secure_lib_libssl}
+		${_secure_lib_libcrypto} ${_secure_lib_libssh} \
+		${_secure_lib_libssl}
 
 .if ${MK_LIBTHR} != "no"
 _lib_libthr=	lib/libthr
 .endif
 
-.if ${MK_RADIUS_SUPPORT} != "no"
-_lib_libradius=	lib/libradius
-.endif
-
 .if ${MK_OFED} != "no"
 _ofed_lib=	contrib/ofed/usr.lib/
 .endif
@@ -1417,7 +1373,6 @@ lib/libopie__L lib/libtacplus__L: lib/libmd__L
 .if ${MK_CDDL} != "no"
 _cddl_lib_libumem= cddl/lib/libumem
 _cddl_lib_libnvpair= cddl/lib/libnvpair
-_cddl_lib_libavl= cddl/lib/libavl
 _cddl_lib_libzfs_core= cddl/lib/libzfs_core
 _cddl_lib= cddl/lib
 cddl/lib/libzfs_core__L: cddl/lib/libnvpair__L
@@ -1569,7 +1524,6 @@ delete-old-files:
 # argument list will get too long. Using .for/.endfor make "loops" will make
 # the Makefile parser segfault.
 	@exec 3<&0; \
-	cd ${.CURDIR}; \
 	${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
 	    -V OLD_FILES -V "OLD_FILES:Musr/share/*.gz:R" | xargs -n1 | \
 	while read file; do \
@@ -1592,8 +1546,7 @@ delete-old-files:
 
 check-old-files:
 	@echo ">>> Checking for old files"
-	@cd ${.CURDIR}; \
-	${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
+	@${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
 	    -V OLD_FILES -V "OLD_FILES:Musr/share/*.gz:R" | xargs -n1 | \
 	while read file; do \
 		if [ -f "${DESTDIR}/$${file}" -o -L "${DESTDIR}/$${file}" ]; then \
@@ -1614,7 +1567,6 @@ delete-old-libs:
 	@echo ">>> Removing old libraries"
 	@echo "${OLD_LIBS_MESSAGE}" | fmt
 	@exec 3<&0; \
-	cd ${.CURDIR}; \
 	${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
 	    -V OLD_LIBS | xargs -n1 | \
 	while read file; do \
@@ -1622,36 +1574,22 @@ delete-old-libs:
 			chflags noschg "${DESTDIR}/$${file}" 2>/dev/null || true; \
 			rm ${RM_I} "${DESTDIR}/$${file}" <&3; \
 		fi; \
-		for ext in debug symbols; do \
-		  if ! [ -e "${DESTDIR}/$${file}" ] && [ -f \
-		      "${DESTDIR}${DEBUGDIR}/$${file}.$${ext}" ]; then \
-			  rm ${RM_I} "${DESTDIR}${DEBUGDIR}/$${file}.$${ext}" \
-			      <&3; \
-		  fi; \
-		done; \
 	done
 	@echo ">>> Old libraries removed"
 
 check-old-libs:
 	@echo ">>> Checking for old libraries"
-	@cd ${.CURDIR}; \
-	${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
+	@${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
 	    -V OLD_LIBS | xargs -n1 | \
 	while read file; do \
 		if [ -f "${DESTDIR}/$${file}" -o -L "${DESTDIR}/$${file}" ]; then \
 			echo "${DESTDIR}/$${file}"; \
 		fi; \
-		for ext in debug symbols; do \
-		  if [ -f "${DESTDIR}${DEBUGDIR}/$${file}.$${ext}" ]; then \
-			  echo "${DESTDIR}${DEBUGDIR}/$${file}.$${ext}"; \
-		  fi; \
-		done; \
 	done
 
 delete-old-dirs:
 	@echo ">>> Removing old directories"
-	@cd ${.CURDIR}; \
-	${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
+	@${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
 	    -V OLD_DIRS | xargs -n1 | \
 	while read dir; do \
 		if [ -d "${DESTDIR}/$${dir}" ]; then \
@@ -1664,8 +1602,7 @@ delete-old-dirs:
 
 check-old-dirs:
 	@echo ">>> Checking for old directories"
-	@cd ${.CURDIR}; \
-	${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
+	@${MAKE} -f ${.CURDIR}/Makefile.inc1 ${.MAKEFLAGS} ${.TARGET} \
 	    -V OLD_DIRS | xargs -n1 | \
 	while read dir; do \
 		if [ -d "${DESTDIR}/$${dir}" ]; then \
@@ -1696,7 +1633,7 @@ DTBOUTPUTPATH= ${KRNLOBJDIR}/${KERNCONF}/
 .if !defined(FDT_DTS_FILE) || empty(FDT_DTS_FILE)
 .if exists(${KERNCONFDIR}/${KERNCONF})
 FDT_DTS_FILE!= awk 'BEGIN {FS="="} /^makeoptions[[:space:]]+FDT_DTS_FILE/ {print $$2}' \
-	'${KERNCONFDIR}/${KERNCONF}' ; echo
+	${KERNCONFDIR}/${KERNCONF} ; echo
 .endif
 .endif
 
diff --git a/sys/amd64/conf/GENERIC b/sys/amd64/conf/GENERIC
index 07f60eb..eaaed07 100644
--- a/sys/amd64/conf/GENERIC
+++ b/sys/amd64/conf/GENERIC
@@ -16,18 +16,23 @@
 # If you are in doubt as to the purpose or necessity of a line, check first
 # in NOTES.
 #
-# $FreeBSD$
+# $FreeBSD: stable/9/sys/amd64/conf/GENERIC 253860 2013-08-01 13:18:47Z marius $
 
 cpu		HAMMER
 ident		GENERIC
 
 makeoptions	DEBUG=-g		# Build kernel with gdb(1) debug symbols
 makeoptions	WITH_CTF=1		# Run ctfconvert(1) for DTrace support
+makeoptions	CAMDEBUG=1
+# # options     CAMDEBUG
 
+options         CLOUDBYTE_11            # For setting PRIV_VFS_GENERATION for CB jails
+options 	BREAK_TO_DEBUGGER
+options 	SW_WATCHDOG     	# Enables software watchdog
 options 	SCHED_ULE		# ULE scheduler
 options 	PREEMPTION		# Enable kernel thread preemption
 options 	INET			# InterNETworking
-options 	INET6			# IPv6 communications protocols
+# options 	INET6			# IPv6 communications protocols
 options 	TCP_OFFLOAD		# TCP offload
 options 	SCTP			# Stream Control Transmission Protocol
 options 	FFS			# Berkeley Fast Filesystem
@@ -62,6 +67,12 @@ options 	SYSVSEM			# SYSV-style semaphores
 options 	_KPOSIX_PRIORITY_SCHEDULING # POSIX P1003_1B real-time extensions
 options 	PRINTF_BUFR_SIZE=128	# Prevent printf output being interspersed.
 options 	KBD_INSTALL_CDEV	# install a CDEV entry in /dev
+#options 	LOCK_PROFILING          # Add profiling for performance tracking
+
+device		dcons			# dconsole driver
+#device		dcons_crom
+
+#device		hwpmc			# Driver (also a loadable module) Enable for profiling
 options 	HWPMC_HOOKS		# Necessary kernel hooks for hwpmc(4)
 options 	AUDIT			# Security event auditing
 options 	MAC			# TrustedBSD MAC Framework
@@ -72,6 +83,9 @@ options 	KDB			# Kernel debugger related code
 options 	KDB_TRACE		# Print a stack trace for a panic
 options 	DDB_CTF			# kernel ELF linker loads CTF data
 
+options 	GDB			# Enabled GDB
+options		DDB			# Enabled DDB
+options 	DDB_NUMSYM
 # Make an SMP-capable kernel by default
 options 	SMP			# Symmetric MultiProcessor Kernel
 
@@ -83,7 +97,7 @@ device		acpi
 device		pci
 
 # Floppy drives
-device		fdc
+#device		fdc
 
 # ATA controllers
 device		ahci		# AHCI-compatible SATA controllers
@@ -103,11 +117,12 @@ options 	AHD_REG_PRETTY_PRINT	# Print register bitfields in debug
 device		esp		# AMD Am53C974 (Tekram DC-390(T))
 device		hptiop		# Highpoint RocketRaid 3xxx series
 device		isp		# Qlogic family
-#device		ispfw		# Firmware for QLogic HBAs- normally a module
+device		ispfw		# Firmware for QLogic HBAs- normally a module
+options         ISP_TARGET_MODE # Target Mode for ISP
 device		mpt		# LSI-Logic MPT-Fusion
-device		mps		# LSI-Logic MPT-Fusion 2
-device		mpr		# LSI-Logic MPT-Fusion 3
+#device		mps		# LSI-Logic MPT-Fusion 2
 #device		ncr		# NCR/Symbios Logic
+device		mpr		# LSI-Logic MPT-Fusion 3
 device		sym		# NCR/Symbios Logic (newer chipsets + those of `ncr')
 device		trm		# Tekram DC395U/UW/F DC315U adapters
 
@@ -125,7 +140,7 @@ device		sa		# Sequential Access (tape etc)
 device		cd		# CD
 device		pass		# Passthrough device (direct ATA/SCSI access)
 device		ses		# Enclosure Services (SES and SAF-TE)
-#device		ctl		# CAM Target Layer
+device		ctl		# CAM Target Layer
 
 # RAID controllers interfaced to the SCSI subsystem
 device		amr		# AMI MegaRAID
@@ -150,8 +165,9 @@ device		aacp		# SCSI passthrough for aac (requires CAM)
 device		aacraid		# Adaptec by PMC RAID
 device		ida		# Compaq Smart RAID
 device		mfi		# LSI MegaRAID SAS
+device		mrsas		# LSI MegaRAID SAS
+device		mfip		# LSI MegaRAID SAS passthrough, requires CAM
 device		mlx		# Mylex DAC960 family
-device		mrsas		# LSI/Avago MegaRAID SAS/SATA, 6Gb/s and 12Gb/s
 #XXX pointer/int warnings
 #device		pst		# Promise Supertrak SX6000
 device		twe		# 3ware ATA RAID
@@ -194,12 +210,13 @@ device		ppi		# Parallel port interface device
 device		puc		# Multi I/O cards and multi-channel UARTs
 
 # PCI Ethernet NICs.
-device		bxe		# Broadcom NetXtreme II BCM5771X/BCM578XX 10GbE
+device		bxe		# Broadcom BCM57710/BCM57711/BCM57711E 10Gb Ethernet
 device		de		# DEC/Intel DC21x4x (``Tulip'')
 device		em		# Intel PRO/1000 Gigabit Ethernet Family
 device		igb		# Intel PRO/1000 PCIE Server Gigabit Family
 device		ixgbe		# Intel PRO/10GbE PCIE Ethernet Family
 device		le		# AMD Am7900 LANCE and Am79C9xx PCnet
+device		oce 		# Emulex 10 Gbit adapters (OneConnect Ethernet)
 device		ti		# Alteon Networks Tigon I/II gigabit Ethernet
 device		txp		# 3Com 3cR990 (``Typhoon'')
 device		vx		# 3Com 3c590, 3c595 (``Vortex'')
@@ -253,30 +270,30 @@ device		sn		# SMC's 9000 series of Ethernet chips
 device		xe		# Xircom pccard Ethernet
 
 # Wireless NIC cards
-device		wlan		# 802.11 support
-options 	IEEE80211_DEBUG	# enable debug msgs
-options 	IEEE80211_AMPDU_AGE # age frames in AMPDU reorder q's
-options 	IEEE80211_SUPPORT_MESH	# enable 802.11s draft support
-device		wlan_wep	# 802.11 WEP support
-device		wlan_ccmp	# 802.11 CCMP support
-device		wlan_tkip	# 802.11 TKIP support
-device		wlan_amrr	# AMRR transmit rate control algorithm
-device		an		# Aironet 4500/4800 802.11 wireless NICs.
-device		ath		# Atheros NICs
-device		ath_pci		# Atheros pci/cardbus glue
-device		ath_hal		# pci/cardbus chip support
-options 	AH_SUPPORT_AR5416	# enable AR5416 tx/rx descriptors
-device		ath_rate_sample	# SampleRate tx rate control for ath
+#device		wlan		# 802.11 support
+#options 	IEEE80211_DEBUG	# enable debug msgs
+#options 	IEEE80211_AMPDU_AGE # age frames in AMPDU reorder q's
+#options 	IEEE80211_SUPPORT_MESH	# enable 802.11s draft support
+#device		wlan_wep	# 802.11 WEP support
+#device		wlan_ccmp	# 802.11 CCMP support
+#device		wlan_tkip	# 802.11 TKIP support
+#device		wlan_amrr	# AMRR transmit rate control algorithm
+#device		an		# Aironet 4500/4800 802.11 wireless NICs.
+#device		ath		# Atheros NICs
+#device		ath_pci		# Atheros pci/cardbus glue
+#device		ath_hal		# pci/cardbus chip support
+#options 	AH_SUPPORT_AR5416	# enable AR5416 tx/rx descriptors
+#device		ath_rate_sample	# SampleRate tx rate control for ath
 #device		bwi		# Broadcom BCM430x/BCM431x wireless NICs.
 #device		bwn		# Broadcom BCM43xx wireless NICs.
-device		ipw		# Intel 2100 wireless NICs.
-device		iwi		# Intel 2200BG/2225BG/2915ABG wireless NICs.
-device		iwn		# Intel 4965/1000/5000/6000 wireless NICs.
-device		malo		# Marvell Libertas wireless NICs.
-device		mwl		# Marvell 88W8363 802.11n wireless NICs.
-device		ral		# Ralink Technology RT2500 wireless NICs.
-device		wi		# WaveLAN/Intersil/Symbol 802.11 wireless NICs.
-device		wpi		# Intel 3945ABG wireless NICs.
+#device		ipw		# Intel 2100 wireless NICs.
+#device		iwi		# Intel 2200BG/2225BG/2915ABG wireless NICs.
+#device		iwn		# Intel 4965/1000/5000/6000 wireless NICs.
+#device		malo		# Marvell Libertas wireless NICs.
+#device		mwl		# Marvell 88W8363 802.11n wireless NICs.
+#device		ral		# Ralink Technology RT2500 wireless NICs.
+#device		wi		# WaveLAN/Intersil/Symbol 802.11 wireless NICs.
+#device		wpi		# Intel 3945ABG wireless NICs.
 
 # Pseudo devices.
 device		loop		# Network loopback
@@ -307,12 +324,12 @@ device		usb		# USB Bus (required)
 #device		udbp		# USB Double Bulk Pipe devices (needs netgraph)
 device		uhid		# "Human Interface Devices"
 device		ukbd		# Keyboard
-device		ulpt		# Printer
+#device		ulpt		# Printer
 device		umass		# Disks/Mass storage - Requires scbus and da
-device		ums		# Mouse
-device		urio		# Diamond Rio 500 MP3 player
+#device		ums		# Mouse
+#device		urio		# Diamond Rio 500 MP3 player
 # USB Serial devices
-device		u3g		# USB-based 3G modems (Option, Huawei, Sierra)
+#device		u3g		# USB-based 3G modems (Option, Huawei, Sierra)
 device		uark		# Technologies ARK3116 based serial adapters
 device		ubsa		# Belkin F5U103 and compatible serial adapters
 device		uftdi		# For FTDI usb serial adapters
@@ -322,32 +339,32 @@ device		uslcom		# SI Labs CP2101/CP2102 serial adapters
 device		uvisor		# Visor and Palm devices
 device		uvscom		# USB serial support for DDI pocket's PHS
 # USB Ethernet, requires miibus
-device		aue		# ADMtek USB Ethernet
-device		axe		# ASIX Electronics USB Ethernet
-device		cdce		# Generic USB over Ethernet
-device		cue		# CATC USB Ethernet
-device		kue		# Kawasaki LSI USB Ethernet
-device		rue		# RealTek RTL8150 USB Ethernet
-device		udav		# Davicom DM9601E USB
+nodevice		aue		# ADMtek USB Ethernet
+nodevice		axe		# ASIX Electronics USB Ethernet
+nodevice		cdce		# Generic USB over Ethernet
+nodevice		cue		# CATC USB Ethernet
+nodevice		kue		# Kawasaki LSI USB Ethernet
+nodevice		rue		# RealTek RTL8150 USB Ethernet
+nodevice		udav		# Davicom DM9601E USB
 # USB Wireless
-device		rum		# Ralink Technology RT2501USB wireless NICs
-device		run		# Ralink Technology RT2700/RT2800/RT3000 NICs.
-device		uath		# Atheros AR5523 wireless NICs
-device		upgt		# Conexant/Intersil PrismGT wireless NICs.
-device		ural		# Ralink Technology RT2500USB wireless NICs
-device		urtw		# Realtek RTL8187B/L wireless NICs
-device		zyd		# ZyDAS zd1211/zd1211b wireless NICs
+#device		rum		# Ralink Technology RT2501USB wireless NICs
+#device		run		# Ralink Technology RT2700/RT2800/RT3000 NICs.
+#device		uath		# Atheros AR5523 wireless NICs
+#device		upgt		# Conexant/Intersil PrismGT wireless NICs.
+#device		ural		# Ralink Technology RT2500USB wireless NICs
+#device		urtw		# Realtek RTL8187B/L wireless NICs
+#device		zyd		# ZyDAS zd1211/zd1211b wireless NICs
 
 # Sound support
-device		sound		# Generic sound driver (required)
-device		snd_cmi		# CMedia CMI8338/CMI8738
-device		snd_csa		# Crystal Semiconductor CS461x/428x
-device		snd_emu10kx	# Creative SoundBlaster Live! and Audigy
-device		snd_es137x	# Ensoniq AudioPCI ES137x
-device		snd_hda		# Intel High Definition Audio
-device		snd_ich		# Intel, NVidia and other ICH AC'97 Audio
-device		snd_uaudio	# USB Audio
-device		snd_via8233	# VIA VT8233x Audio
+#device		sound		# Generic sound driver (required)
+#device		snd_cmi		# CMedia CMI8338/CMI8738
+#device		snd_csa		# Crystal Semiconductor CS461x/428x
+#device		snd_emu10kx	# Creative SoundBlaster Live! and Audigy
+#device		snd_es137x	# Ensoniq AudioPCI ES137x
+#device		snd_hda		# Intel High Definition Audio
+#device		snd_ich		# Intel, NVidia and other ICH AC'97 Audio
+#device		snd_uaudio	# USB Audio
+#device		snd_via8233	# VIA VT8233x Audio
 
 # VirtIO support
 device		virtio		# Generic VirtIO bus (required)
diff --git a/sys/nlm/nlm_prot_clnt.c b/sys/nlm/nlm_prot_clnt.c
index a268e63..6265551 100644
--- a/sys/nlm/nlm_prot_clnt.c
+++ b/sys/nlm/nlm_prot_clnt.c
@@ -7,6 +7,7 @@
 #include <sys/lock.h>
 #include <sys/mutex.h>
 #include <sys/systm.h>
+#include <sys/syscallsubr.h>
 
 #include <nlm/nlm_prot.h>
 #include <sys/cdefs.h>
@@ -17,6 +18,9 @@ __RCSID("$NetBSD: nlm_prot.x,v 1.6 2000/06/07 14:30:15 bouyer Exp $");
 #endif /* not lint */
 __FBSDID("$FreeBSD$");
 
+static int nlm_debug_delay_granted_res = 0;
+SYSCTL_INT(_debug, OID_AUTO, nlm_debug_delay_granted_res, CTLFLAG_RW, &nlm_debug_delay_granted_res, 0, "");
+
 enum clnt_stat 
 nlm_sm_notify_0(struct nlm_sm_status *argp, void *clnt_res, CLIENT *clnt, struct rpc_callextra *ext, struct timeval timo)
 {
@@ -326,6 +330,13 @@ nlm4_unlock_res_4(nlm4_res *argp, void *clnt_res, CLIENT *clnt, struct rpc_calle
 enum clnt_stat 
 nlm4_granted_res_4(nlm4_res *argp, void *clnt_res, CLIENT *clnt, struct rpc_callextra *ext, struct timeval timo)
 {
+	if (nlm_debug_delay_granted_res > 0) {
+		struct timespec rqt = { 
+								.tv_sec = nlm_debug_delay_granted_res, 
+								.tv_nsec = 0
+							  };
+		(void) kern_nanosleep(curthread, &rqt, NULL);
+	}
 	return (CLNT_CALL_EXT(clnt, ext, NLM4_GRANTED_RES,
 		(xdrproc_t) xdr_nlm4_res, (caddr_t) argp,
 		(xdrproc_t) xdr_void, (caddr_t) clnt_res,
